example.js

```diff
@@ -73,6 +73,18 @@ const { prompt } = require('./');
                 { title: 'Gibson' },
                 { title: 'Grant' },
             ]
+        },
+        {
+            type: 'number',
+            name: 'prompt',
+            message: 'This will be overridden',
+            onRender() {
+                this.no = (this.no || 0) + 1;
+                this.msg = `Enter a number (e.g. ${this.no})`;
+                new Promise(resolve => setTimeout(resolve, 1000)).then(() => {
```

Comment:
Why do you wrap this in a promise, I think this is fine no?
```js
setTimeout(() => {
  this.no+=1;
  this.render();
}, 1000);
```

That being said the example isn't that good in my opinion because you end up creating a new `setTimeout` on every keypress which makes the counter go a bit crazy. 

A few open questions:
- What about exposing the color formatting utils as param, in case people want to add colors to the new prompt?
- How would one go about having a fixed interval update, as you'd expect from a render loop?